disktools.py

```diff
@@ -1267,3 +1267,60 @@ def getMdNodes():
     except IOError:
         pass
     return nodes
+
+
+def readCmdlineMounts():
+    class Mount:
+        def __init__(self, dev, mountpoint, options=None, fstype=None):
+            self.dev = dev
+            self.mountpoint = mountpoint
+            self.options = options
+            self.fstype = fstype
+
+        def __repr__(self):
+            return "Mount(%s, %s, %s, %s)" % (self.dev, self.mountpoint, self.fstype, self.options)
+
+        def mount(self):
+            util.mount(self.dev, self.mountpoint,
+                  self.options if self.options is None else [self.options], self.fstype)
+
+        def umount(self):
+            util.umount(self.mountpoint)
+
+    def quoteDiskLabel(label):
+        return label.replace('\\', '\\x5c').replace(' ', '\\x20')
+
+    mounts = []
+    with open('/proc/cmdline') as f:
+        for m in [x for x in f.read().rstrip().split(' ') if x.startswith('mount=')]:
+            m = m.split(':')
+            if len(m) < 2:
+                continue
+            dev = m[0][6:]
+            fstype = None
+            options = None
+            if len(m) >= 4:
+                fstype = m[1]
+                options = m[2]
+                mnt = m[3]
+            elif len(m) >= 3:
+                fstype = m[1]
+                mnt = m[2]
+            else:
+                mnt = m[1]
+            if dev.startswith("LABEL="):
+                dev = '/dev/disk/by-label/' + quoteDiskLabel(dev[6:])
```

Comment:
Why not call `blkid -t` like the old code did? That way it also handles `UUID=...` as well, like it previously did (and doesn't rely on us getting the arbitrary quoting correct).

Reply:
I don't think we supposed `UUID=` or if we did the support was removed. The script only check for `LABEL=`. One note is that the label in the kernel will be available after the device can be read; that means that the combination of `LABEL=` and USB still setting is currently (even before this commit) not working.

But probably nice to avoid regression and use `blkid`.